cmd/scip/main_test.go

```diff
@@ -101,3 +104,116 @@ func TestSCIPSnapshots(t *testing.T) {
 		return snapshots
 	})
 }
+
+func unwrap[T any](v T, err error) func(*testing.T) T {
+	return func(t *testing.T) T {
+		require.NoError(t, err)
+		return v
+	}
+}
+
+func TestSCIPTests(t *testing.T) {
+	cwd := unwrap(os.Getwd())(t)
+	testDir := filepath.Join(cwd, "tests", "test_cmd")
+	testPaths := unwrap(os.ReadDir(testDir))(t)
+	require.Truef(t, len(testPaths) >= 1, "Expected at least one test case in directory: %v", testDir)
+
+	os.Setenv("NO_COLOR", "1")
+	t.Cleanup(func() {
+		os.Unsetenv("NO_COLOR")
+	})
+
+	type TestCase struct {
+		dir        string
+		passOutput autogold.Value
+		failOutput autogold.Value
+	}
+
+	// To update the snapshot values, run 'go test ./cmd/scip -update'.
+	testCases := []TestCase{
+		{"roles",
+			autogold.Expect("✓ passes.repro (3 assertions)\n"),
+			autogold.Expect(`✗ fails-wrong-role.repro
```

Comment:
@matthewnitschke-wk I've added snapshot tests here for checking the output, so that we don't accidentally regress something (earlier, this test was just checking for pass/fail instead of checking the output).